### ISSUE TITLE
Could not open required defaults file: /etc/mysql-<service>/my.cnf

### ISSUE THREAD

**Issue by binpath** [chef-cookbooks/mysql#449]
### Cookbook version
8.0.1
### Chef-client version
Chef Development Kit Version: 0.12.0
chef-client version: 12.8.1
berks version: 4.3.0
kitchen version: 1.6.0
### Platform Details
Ubuntu-14.04 (AWS)

### Scenario:
Creating the learn chef tutorial of chef (https://learn.chef.io/manage-a-web-app/ubuntu/configure-mysql/

STDOUT: Installing MySQL system tables...
       STDERR: Could not open required defaults file: /etc/mysql-fairprice/my.cnf
       Fatal error in defaults handling. Program aborted
       ---- End output of "bash"  "/tmp/chef-script20160810-23221-13lj0j4" ----
       Ran "bash"  "/tmp/chef-script20160810-23221-13lj0j4" returned 141
       [2016-08-10T01:28:41+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

I had to put package_name, and package_version because of an error also, that the version is not found.

### Steps to Reproduce:
you using this cookbook or any resources it includes?]
in the recipe database.rb

mysql_client 'fairprice_client' do
  package_name node['wrapper_mysql']['client_package_name']
  package_version node['wrapper_mysql']['client_package_version']
  action :create
end

mysql_service 'fairprice' do
  data_dir node['wrapper_mysql']['data_dir']
  initial_root_password node['wrapper_mysql']['initial_root_password']
  package_name node['wrapper_mysql']['server_package_name']
  package_version node['wrapper_mysql']['server_package_version']
  run_group 'mysql'
  run_user 'mysql'
  action [:create, :start]
end
### Expected Result:
Installed the instance mysql service properly like the learning chef tutorial

### Actual Result:

------------------
           # Declared in /tmp/kitchen/cache/cookbooks/fp_mysql/recipes/database.rb:11:in `from_file'
           
           mysql_service("fairprice") do
             action [:create, :start]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :mysql_service
             cookbook_name "wrapper_mysql"
             recipe_name "database"
             data_dir "/usr/mysql/data/"
             initial_root_password "************"
             package_name "mysql-server-5.6"
             package_version "5.6.31-0ubuntu0.14.04.2"
             run_group "ubuntu"
             run_user "ubuntu"
             install_method "auto"
             service_manager "auto"
           end
           
           Platform:
           ---------
           x86_64-linux
           
       
       Running handlers:
       [2016-08-10T01:28:39+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2016-08-10T01:28:39+00:00] ERROR: Exception handlers complete
       Chef Client failed. 2 resources updated in 04 seconds
       [2016-08-10T01:28:39+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2016-08-10T01:28:39+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2016-08-10T01:28:39+00:00] ERROR: mysql_service[fairprice] (fp_mysql::database line 11) had an error: Mixlib::ShellOut::ShellCommandFailed: mysql_service_manager_upstart[fairprice] (/tmp/kitchen/cache/cookbooks/mysql/libraries/mysql_service.rb line 50) had an error: Mixlib::ShellOut::ShellCommandFailed: bash[fairprice initial records] (/tmp/kitchen/cache/cookbooks/mysql/libraries/mysql_service_base.rb line 111) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0, 1, 2], but received '141'
       ---- Begin output of "bash"  "/tmp/chef-script20160810-23221-13lj0j4" ----
       STDOUT: Installing MySQL system tables...
       STDERR: Could not open required defaults file: /etc/mysql-fairprice/my.cnf
       Fatal error in defaults handling. Program aborted
       ---- End output of "bash"  "/tmp/chef-script20160810-23221-13lj0j4" ----
       Ran "bash"  "/tmp/chef-script20160810-23221-13lj0j4" returned 141
       [2016-08-10T01:28:41+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

**Comment by AneudyM** [comment]
@binpath I am facing the same issue and it seems to be due to AppArmor. Take a look at this:

````
mininode@mini-node:~$ dmesg | grep 'profile="/usr/sbin/mysqld"'
[ 2327.642068] type=1400 audit(1471269366.907:33): apparmor="DENIED" operation="open" profile="/usr/sbin/mysqld" name="/etc/mysql-crossover/my.cnf" pid=3362 comm="mysqld" requested_mask="r" denied_mask="r" fsuid=0 ouid=105
````

This is telling you the following:

- `apparmor="DENIED"`: AppArmor denied/prevented something from happening
- `operation="open"`: the operation AppArmor denied/prevented from happening
- `profile="/usr/sbin/mysqld"`: the profile that made AppArmor deny the operation
- `name="/etc/mysql-crossover/my.cnf"`: the name of the resource upon which the operation was supposed to be performed
- `pid=3362`: the PID of the process that tried to perform the operation
- `comm="mysqld"`: the command or process that tried to perform the operation
- `requested_mask="r"`: the requested action the mysqld process wanted to perform ("r" read the resource)
- `denied_mask="r"`: the denied action AppArmor stopped mysqld from doing ("r" read the resource)

**Comment by WesleyE** [comment]
It is indeed Apparmor. It is fixed in pr #453, which works for me. Hope it gets merged soon.

**Comment by binpath** [comment]
Thanks @WesleyE!
I have used another distro and thats solved my issue, but when it's going to be merged?

**Comment by someara** [comment]
Fixed up in 8.0.2
